### DISCUSSION TITLE
In v4.0.0 Connection 'transaction' event emitter does not include 'Amount' property on tx_json - what should I use?

### DISCUSSION THREAD

**Discussion by MichaelAJay** [XRPLF/xrpl.js#2828] [Q&A] [answered]
In v2.14.0, the Connection 'transaction' event emitted an object which had transaction.Amount.
In v4.0.0, the Connection 'transaction' event emits an object which has tx_json instead of transaction - that's fine. But it's altogether missing 'Amount'.

The 'Payment' interface (src/models/transactions/payment.ts) that I'm looking at indicates that 'Amount' should exist on tx_json (which I'm taking to be an object which conforms to the Payment interface).

According to that documentation, 'Amount' may be used in partial payment scenarios - so am I better using 'meta.delivered_amount'?

**Comment by khancode** [comment]
Yes, `delivered_amount` is now the one to use for v4.0.0; this change is part of rippled API v2 - https://github.com/XRPLF/rippled/pull/4733

**Reply by mvadari** [reply, marked as answer]
If you specifically want the `Amount` value, it has been renamed to `DeliverMax`.

The full API changelog is available here: https://github.com/xrplf/rippled/blob/develop/API-CHANGELOG.md#api-version-2

You can also switch back to API v1 with `client.apiVersion = 1` or specifying the `api_version` in the request.
